docs/adr/0019-use-1password-for-sharing-secrets.md

```diff
@@ -0,0 +1,96 @@
+
+## Decision Outcome
+
+Chosen Alternative: Use a 1Password vault for sharing secrets.
+
+* 1Password allows us to control access to the secrets repository on a per-user
+  basis, and updates to any secret will be available to other users on the
+  vault immediately.  With a separate vault within 1Password, we need not share
+  secrets with other Truss personnel who are not assigned to the project.
+* 1Password also has familiar and user-friendly graphical interfaces via the
+  web and a multi-platform desktop application.
+* Access to secrets can be revoked as quickly and as easily as it was granted.
+
+Consequences:
+
+* We will need to create a vault, which takes time and costs money.
+* Some synchronization will be required on how to use the vault and what
+  secrets to store in it.
+* Secrets that are required by the easi-backend app will have to be updated in
+  SSM anyway, which means changing them requires updating the value in two
+  places (1Password and SSM).
+
+## Pros and Cons of the Alternatives
+
+### Email
+
+* `+` Fast.
+* `+` Some of our peers share secrets with us using email, so forwarding the
+  emails is an easy and straightforward way to distribute secrets.
+* `-` Poor access controls: once an email is sent, its contents cannot be recalled.
+* `-` Emails can be easily shared with a wide audience inadvertently, resulting
+  in a pervasive risk of uncontrolled disclosure.
+* `-` If we have to manage any secrets that cannot be rotated, there will be no
+  way to guarantee that sent emails are burned after reading: rather, they will
+  likely live on in the archived email of whoever received them.
+* `-` There would be no central source of truth on what the current secrets
+  are, which complicates onboarding/off-boarding, engineers returning from
+  vacation, etc.
+
+### Slack DMs
+
+* `+` It's easy to find people in Slack, and we all use it.
+* `+` Only two people are in a DM thread. Unlike email, it is not easy to share
+  contents of a DM inadvertently.
+* `-` Due to CMS record-keeping policies, DMs cannot be deleted and may be
+  subject to subpoena or FOIA requests, which makes them public records.
+* `-` There is no central source of truth.
+
+### 1Password
+
+* `+` Per-user access controls.
+* `+` We can restrict the vault to project engineers according to the principle
+  of least access.
+* `+` The 1Password vault is a central source of truth on secrets.
+* `-` We have to create a vault, which costs money. May require CORS approval.
+* `-` Some secrets will have to be updated in 1Password and SSM when they
+  change, such as API keys.
+
+### AWS SSM / ParameterStore
+
+_See also:_ [Use AWS SSM Parameter Store for Config/Secrets management](./0008-ssm-for-configsecrets.md)
+
+* `+` Provides a _single_ source of truth: Updates in SSM are automatically
+  shared with the easi-backend app.
+* `+` We already depend on ParameterStore to store secrets for the app, so
+  there is no new point of potential security failure.
```

Comment:
Given the way this second point is phrased, it's unclear why AWS SSM / ParameterStore is not the chosen alternative.

Reply:
This is a `+` about how SSM meets the security decision driver. Overall, SSM doesn't meet the user friendliness decision driver.

Reply:
Ah, okay, I think I initially misinterpreted what the second part of this bullet item was conveying.